src/scheduler/mod.rs

```diff
@@ -328,6 +340,45 @@ impl PerCoreScheduler {
 			.as_u64();
 	}
 
+	pub fn set_current_task_priority(&mut self, prio: Priority) {
+		irqsave(|| {
+			info!("Change priority of the current task");
+			self.current_task.borrow_mut().prio = prio;
+		});
+	}
+
+	pub fn set_priority(&mut self, id: TaskId, prio: Priority) -> Result<(), ()> {
+		info!("Chabge priority of task {} to priority {}", id, prio);
+
+		irqsave(|| {
+			let task = get_task_handle(id).ok_or(())?;
+
+			#[cfg(feature = "smp")]
+			if task.get_core_id() == self.core_id {
+				if self.current_task.borrow().id == task.get_id() {
+					self.current_task.borrow_mut().prio = prio;
+				} else {
+					self.ready_queue
+						.set_priority(task, prio)
+						.expect("Do not find valid task in ready queue");
+				}
+			} else {
+				warn!("Have to change the priority on another core");
+			}
+
+			#[cfg(not(feature = "smp"))]
+			if self.current_task.borrow().id == task.get_id() {
```

Comment:
We can reduce code duplication:

```suggestion
			if cfg!(feature = "smp") && task.get_core_id() != self.core_id {
				warn!("Have to change the priority on another core");
			} else if self.current_task.borrow().id == task.get_id() {
```